StationeersLaunchPad/LoadedMod.cs

```diff
@@ -79,146 +116,129 @@ public UniTask FindEntrypoints()
-          {
-            if (!this.SMEntryTypes.Contains(type))
-              this.SMEntryTypes.Add(type);
-          }
-        }
-        foreach (var exportSettings in this.Exports)
-        {
-          var prefab = exportSettings._startupPrefab;
-          if (prefab != null)
-            this.EntryPrefabs.Add(prefab);
-        }
-        this.BepinexEntryTypes.AddRange(ModLoader.FindBepinexEntrypoints(this.Assemblies));
+          this.SMEntryTypes.AddRange(ModLoader.FindAnyStationeersModsEntrypoints(this.Assemblies));
+        else
+          this.SMEntryTypes.AddRange(ModLoader.FindExplicitStationeersModsEntrypoints(this.Assemblies));
 
+        if (this.SMEntryTypes.Count == 0)
+          this.SMEntryTypes.AddRange(ModLoader.FindExportSettingsClassEntrypoints(this.Assemblies, this.Exports));
+
+        this.BepInExEntryTypes.AddRange(ModLoader.FindBepInExEntrypoints(this.Assemblies));
         this.DefaultEntryTypes.AddRange(ModLoader.FindDefaultEntrypoints(this.Assemblies));
+
+        this.Logger.LogDebug("Found Entrypoints");
       });
     }
 
-    public void PrintEntrypoints()
+    public async UniTask PrintEntrypoints()
     {
       // getting prefab names fails on a thread in the debug player, so just print all the entrypoints after we finish
       foreach (var type in this.SMEntryTypes)
-        this.Logger.Log($"- StationeersMods {type.FullName}");
+        this.Logger.Log($"- StationeersMods Entry {type.FullName}");
       foreach (var prefab in this.EntryPrefabs)
-        this.Logger.Log($"- prefab {prefab.name}");
-      foreach (var type in this.BepinexEntryTypes)
-        this.Logger.Log($"- Bepinex {type.FullName}");
-      foreach (var (type, _) in this.DefaultEntryTypes)
-        this.Logger.Log($"- Default {type.FullName}");
+        this.Logger.Log($"- Prefab Entry {prefab.name}");
+      foreach (var type in this.BepInExEntryTypes)
+        this.Logger.Log($"- BepInEx Entry {type.FullName}");
+      foreach (var (type, method) in this.DefaultEntryTypes)
+        this.Logger.Log($"- Default Entry {type.FullName}");
     }
 
-    public void LoadEntrypoints()
-    {
-      this.Logger.Log("Loading Entrypoints");
 
-      // StationeersMods tagged ModBehaviour/StartupClass/StartupPrefab
-      var modBehaviours = new List<ModBehaviour>();
-      if (this.SMEntryTypes.Count > 0)
-      {
-        var gameObj = new GameObject();
-        GameObject.DontDestroyOnLoad(gameObj);
-        foreach (var type in this.SMEntryTypes)
-          gameObj.AddComponent(type);
-        modBehaviours.AddRange(gameObj.GetComponents<ModBehaviour>());
-      }
-      foreach (var prefab in this.EntryPrefabs)
-      {
-        var gameObj = GameObject.Instantiate(prefab);
-        GameObject.DontDestroyOnLoad(gameObj);
-        modBehaviours.AddRange(gameObj.GetComponents<ModBehaviour>());
-      }
-      foreach (var modBehaviour in modBehaviours)
+    public async UniTask LoadEntrypoints()
```

Comment:
same here on no async. its gotta be on the main thread for creating game objects.

also super extra nit: don't need commented out code. git remembers all